tests/bluetooth/bsim_bt/bsim_test_adv_resume/src/central.c

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "bs_bt_utils.h"
+#include "zephyr/bluetooth/addr.h"
+#include "zephyr/bluetooth/conn.h"
+
+#include <stdint.h>
+
+#include <zephyr/bluetooth/bluetooth.h>
+
+void adv_new_ident(void)
+{
+	/* create a new identity:
+	 * the stack disallows connecting to the same peer twice
+	 */
+	int id_a = bt_id_create(NULL, NULL);
+
+	ASSERT(id_a >= 0, "bt_id_create id_a failed (err %d)\n", id_a);
+	advertise_connectable(id_a, 0);
+}
+
+void central_procedure(void)
+{
+	bs_bt_utils_setup();
+	k_msleep(1000);
```

Comment:
Why is this sleep needed?

Reply:
So the logs from bootup and HCI init are not interspersed with the logs from the test suite.

Reply:
I'm not sure I follow. Is this just to have the logs more separated? 

IMO it would be much better to wait on some event, than a timer with a fairly arbitrary value (1000ms) in that case. 

At the very least such things should be documented/commented, but I don't really understand why you'd need the logs separated in the first place.

Reply:
well it's just easier to parse. I highlighted in red what I don't care about in the test:
with the wait: 
![image](https://user-images.githubusercontent.com/68428270/207867940-c484566e-94a6-4f87-81cf-108fc2dd5418.png)
without waiting
![image](https://user-images.githubusercontent.com/68428270/207868136-d6c264c4-800b-4e8d-98a3-37abc9ff1f0c.png)

I can of course remove it if you don't like it.

Reply:
Yeah, I see what you mean. 

This is caused by the logs being delayed (offloaded), and that the test uses `printk`.

Two possible (non-timer) solutions are
1) Use `CONFIG_LOG_MODE_IMMEDIATE` (may slightly affect the timing of the stack)
2) Use the logging system in the test

The same argument with wanting to separate the logs and the `printk` could be said for e.g. the shell implementation as well. Furthermore, adding a timer here, only helps this the init; any action you do in the test will have the same issue if you enabled any debug logging.

Reply:
Yeah I just removed the wait. For debugging, we can always convert all the printk's to LOG_INFs. Don't need to address it now.